### PR TITLE
Remove duplicate boring classes

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

These classes were added to the repo with #12795 

Which means we have duplicate implementation. This is blocking #12614

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [n/a] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @justusschock @awaelchli @rohitgr7 @borda @akihironitta

### CODE REVIEW

File: tests/helpers/boring_model.py [hunk 1]
```diff
@@ -11,12 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Optional
-
 import torch
-from torch.utils.data import DataLoader, Dataset, IterableDataset, Subset
+from torch.utils.data import Dataset, IterableDataset
+
+from pytorch_lightning.demos.boring_classes import BoringDataModule, BoringModel, ManualOptimBoringModel, RandomDataset
```
Comment (carmocca) [inline tests/helpers/boring_model.py:17]: This import could be removed so all tests point directly to these. But that will be a huge PR

Comment (rohitgr7) [reply tests/helpers/boring_model.py:17]: why are we keeping the one in demos? is it relevant to the package?

Comment (carmocca) [reply tests/helpers/boring_model.py:17]: My best guess is that @williamFalcon wanted a class in the examples that could be directly imported by the users (hence the need to live in the package)

**Review by otaj — APPROVED** [review]
I hope all of the removed ones have a counterpart in `pytorch_lightning.demos` :D
